src/libraries/System.Net.Http/src/System/Net/Http/BrowserHttpHandler/BrowserHttpHandler.cs

```diff
@@ -212,37 +212,27 @@ private static async Task<WasmFetchResponse> CallFetch(HttpRequestMessage reques
-                    if (request.Content is StringContent)
+                    bool streamingEnabled = false;
+                    if (BrowserHttpInterop.SupportsStreamingRequest())
                     {
-                        string body = await request.Content.ReadAsStringAsync(cancellationToken).ConfigureAwait(true);
+                        request.Options.TryGetValue(EnableStreamingRequest, out streamingEnabled);
+                    }
+
+                    if (streamingEnabled)
+                    {
+                        Stream stream = await request.Content.ReadAsStreamAsync(cancellationToken).ConfigureAwait(true);
                         cancellationToken.ThrowIfCancellationRequested();
 
-                        promise = BrowserHttpInterop.Fetch(uri, headerNames.ToArray(), headerValues.ToArray(), optionNames, optionValues, abortController, body);
+                        ReadableStreamPullState pullState = new ReadableStreamPullState(stream, cancellationToken);
+
+                        promise = BrowserHttpInterop.Fetch(uri, headerNames.ToArray(), headerValues.ToArray(), optionNames, optionValues, abortController, ReadableStreamPull, pullState);
                     }
                     else
                     {
-                        bool streamingEnabled = false;
-                        if (BrowserHttpInterop.SupportsStreamingRequest())
-                        {
-                            request.Options.TryGetValue(EnableStreamingRequest, out streamingEnabled);
-                        }
-
-                        if (streamingEnabled)
-                        {
-                            Stream stream = await request.Content.ReadAsStreamAsync(cancellationToken).ConfigureAwait(true);
-                            cancellationToken.ThrowIfCancellationRequested();
-
-                            ReadableStreamPullState pullState = new ReadableStreamPullState(stream, cancellationToken);
-
-                            promise = BrowserHttpInterop.Fetch(uri, headerNames.ToArray(), headerValues.ToArray(), optionNames, optionValues, abortController, ReadableStreamPull, pullState);
-                        }
-                        else
-                        {
-                            byte[] buffer = await request.Content.ReadAsByteArrayAsync(cancellationToken).ConfigureAwait(true);
-                            cancellationToken.ThrowIfCancellationRequested();
+                        byte[] buffer = await request.Content.ReadAsByteArrayAsync(cancellationToken).ConfigureAwait(true);
+                        cancellationToken.ThrowIfCancellationRequested();
 
-                            promise = BrowserHttpInterop.Fetch(uri, headerNames.ToArray(), headerValues.ToArray(), optionNames, optionValues, abortController, buffer);
-                        }
+                        promise = BrowserHttpInterop.Fetch(uri, headerNames.ToArray(), headerValues.ToArray(), optionNames, optionValues, abortController, buffer);
```

Comment:
What would happen to encoding UTF8/UTF16/WTF16? 
Right now it's JS interop doing the conversion to JS string and the browser is doing the rest.
https://github.com/dotnet/runtime/blob/f6b6415119b4d6d4421b3186a6a8aaa4803af969/src/mono/wasm/runtime/strings.ts#L71-L78

What would happen to `Content-Encoding` header ?

Reply:
The `Content-Encoding` isn't set automatically and is usually for `gzip` / `br` etc. Or am I wrong?

The string is getting encoded with the encoding specified by the constructor of `StringContent` and stored as bytes because `StringContent` actually inherits from `ByteArrayContent`.

https://github.com/dotnet/runtime/blob/f6b6415119b4d6d4421b3186a6a8aaa4803af969/src/libraries/System.Net.Http/src/System/Net/Http/StringContent.cs#L56-L60

https://github.com/dotnet/runtime/blob/f6b6415119b4d6d4421b3186a6a8aaa4803af969/src/libraries/System.Net.Http/src/System/Net/Http/HttpContent.cs#L25